File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.go
func specialArray(nums []int) int {
    slices.Sort(nums)
    idx := 0
    for x := range(nums[len(nums) - 1] + 1) {
        for x > nums[idx] {
            idx++
        }
        if x == len(nums) - idx {
            return x
        }
    }
    return -1
}

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
auto init = [](){
    ios::sync_with_stdio(0), cin.tie(0), cout.tie(0);
    return 'c';
}();
class Solution {
public:
    int specialArray(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int currNumIdx = 0;
        int x = 0;
        while (currNumIdx < nums.size()) {
            if (x > nums[currNumIdx]) {
                ++currNumIdx;
                continue;
            }
            if (x == nums.size() - currNumIdx) {
                return x;
            }
            ++x;
        }
        return -1;
    }
};